ra-optimisation/search-tree-rao/src/test/java/com/farao_community/farao/search_tree_rao/search_tree/algorithms/SearchTreeBloomerTest.java

```diff
@@ -217,128 +225,164 @@ void testRemoveCombinationsWhichExceedMaxNumberOfRaPerTso() {
+        assertEquals(9, filteredNaCombination.size()); // no combination filtered
 
         // max 2 TSOs
-        bloomer = new SearchTreeBloomer(network, Integer.MAX_VALUE, 2, null, null, false, 0, new ArrayList<>());
+        bloomer = new SearchTreeBloomer(network, Integer.MAX_VALUE, 2, null, null, false, 0, new ArrayList<>(), pState);
         filteredNaCombination = bloomer.removeCombinationsWhichExceedMaxNumberOfTsos(naCombinations, previousLeaf);
 
         assertEquals(7, filteredNaCombination.size());
-        assertFalse(filteredNaCombination.contains(comb2BeNl)); // one combination filtered
+        assertFalse(filteredNaCombination.containsKey(comb2BeNl)); // one combination filtered
 
         // max 1 TSO
-        bloomer = new SearchTreeBloomer(network, Integer.MAX_VALUE, 1, null, null, false, 0, new ArrayList<>());
+        bloomer = new SearchTreeBloomer(network, Integer.MAX_VALUE, 1, null, null, false, 0, new ArrayList<>(), pState);
         filteredNaCombination = bloomer.removeCombinationsWhichExceedMaxNumberOfTsos(naCombinations, previousLeaf);
 
         assertEquals(3, filteredNaCombination.size());
-        assertTrue(filteredNaCombination.contains(indFr2));
-        assertTrue(filteredNaCombination.contains(indFrDe));
-        assertTrue(filteredNaCombination.contains(comb2Fr));
+        assertTrue(filteredNaCombination.containsKey(indFr2));
+        assertTrue(filteredNaCombination.containsKey(indFrDe));
+        assertTrue(filteredNaCombination.containsKey(comb2Fr));
+
+        // check booleans in hashmap -> max 2 TSOs
+        Leaf leaf = Mockito.mock(Leaf.class);
+        bloomer = new SearchTreeBloomer(network, Integer.MAX_VALUE, 2, new HashMap<>(), new HashMap<>(), false, 0, new ArrayList<>(), pState);
+
+        Mockito.when(leaf.getActivatedNetworkActions()).thenReturn(Set.of(naFr1));
+        Mockito.when(leaf.getActivatedRangeActions(Mockito.any(State.class))).thenReturn(Set.of(raBe1));
+
+        Map<NetworkActionCombination, Boolean> naToRemove = bloomer.removeCombinationsWhichExceedMaxNumberOfTsos(naCombinations, leaf);
+        Map<NetworkActionCombination, Boolean> expectedResult = Map.of(indFr2, false, indBe1, false, indNl1, true, indFrDe, false, comb2Fr, false, comb3Be, false, comb2FrNl, true);
+        assertEquals(expectedResult, naToRemove);
     }
 
     @Test
     void testRemoveNetworkActionsFarFromMostLimitingElement() {
 
         // arrange naCombination list
-        List<NetworkActionCombination> naCombinations = List.of(indFr2, indDe1, indBe1, indNl1, indNlBe, indFrDe, indDeNl, comb3Be, comb2De, comb2FrDeBe, comb2BeNl);
+        List<NetworkActionCombination> listOfNaCombinations = List.of(indFr2, indDe1, indBe1, indNl1, indNlBe, indFrDe, indDeNl, comb3Be, comb2De, comb2FrDeBe, comb2BeNl);
+        Map<NetworkActionCombination, Boolean> naCombinations = new HashMap<>();
+        listOfNaCombinations.forEach(na -> naCombinations.put(na, false));
```

Comment:
could you try to make a unit test associated to the functionality you fixed ? 
and more complicated tests with naCombinations that doesn't have only false values

Reply:
What do you mean by the functionality we fixed ? Do you mean a complete test that goes through SearchTree and removes RangeActions if the boolean is True ?

For the second part, we already implemented unit tests that have true values  &#x1F609;